src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -515,19 +515,22 @@ protected function deleteCollection(Document $database, Document $collection, Do
         $databaseId = $database->getId();
         $databaseInternalId = $database->getInternalId();
 
-        $relationships = \array_filter(
-            $collection->getAttribute('attributes'),
-            fn ($attribute) => $attribute['type'] === Database::VAR_RELATIONSHIP
-        );
+        /**
+         * Related collections relating to current collection
+         */
+        $attributes = $dbForProject->find('attributes', [
+            Query::equal('databaseInternalId', [$databaseInternalId]),
+            Query::equal('type', [Database::VAR_RELATIONSHIP]),
+            Query::notEqual('collectionInternalId', $collectionInternalId),
+            Query::contains('options', ['"relatedCollection":"'. $collectionId .'"']),
+            Query::limit(PHP_INT_MAX)
+        ]);
 
-        foreach ($relationships as $relationship) {
-            if (!$relationship['twoWay']) {
-                continue;
-            }
-            $relatedCollection = $dbForProject->getDocument('database_' . $databaseInternalId, $relationship['relatedCollection']);
-            $dbForProject->deleteDocument('attributes', $databaseInternalId . '_' . $relatedCollection->getInternalId() . '_' . $relationship['twoWayKey']);
-            $dbForProject->purgeCachedDocument('database_' . $databaseInternalId, $relatedCollection->getId());
-            $dbForProject->purgeCachedCollection('database_' . $databaseInternalId . '_collection_' . $relatedCollection->getInternalId());
+        foreach ($attributes as $attribute) {
+            $dbForProject->deleteDocument('attributes', $attribute->getId());
+            Console::success('Deleted document "' . $attribute->getId() . '" related collection successfully');
+            $dbForProject->purgeCachedDocument('database_' . $databaseInternalId, $attribute->getAttribute('collectionId'));
+            $dbForProject->purgeCachedCollection('database_' . $databaseInternalId . '_collection_' . $attribute->getAttribute('collectionInternalId'));
         }
```

Comment:
We can use `deleteByGroup` to make this a bit cleaner

Reply:
Hmm it looks like we can't since in the callback I will need to clean cache using 
$attribute->getAttribute('collectionId'))
which only the query result will give me

Reply:
The document (attribute in this case) is passed to the callback as a parameter

Reply:
Niiice Changed :)